### ISSUE TITLE
cannot install the correct poa for ComputeConsensus 

### ISSUE THREAD

**Issue by alicecsq** [ucagenomix/sicelore#33]
Hi, I have similar issues using conda poa=2.0=h516909a_2 to compute consensus. It stalled for a long time while I run ./quickrun.sh and also my own data.

I removed poa from my conda environment. When I tried `cd sicelore` and then `./quickrun.sh.`

<img width="451" alt="Picture 1" src="https://user-images.githubusercontent.com/51415436/115562999-11ec9700-a285-11eb-8eee-6014d433af6f.png">

 It seems like it is using poa from /usr/bin/poa. I added java, run  `./quickrun.sh.` again, and it still stalled on me. 

I cannot figure out how to install the correct poa, e.g. bio-pipeline/poaV2, and add it to my executable path to run sicelore ComputeConsensus properly.

I am wondering if you could help with this issue. Thank you!

**Comment by ucagenomix** [comment]
Hi,
it seems that this is java that is not found not poa. The command to find java is "which $JAVA_HOME"
could you export the JAVA_HOME variable to indicate where is Java executable, it should works,
Note that you could give a try to quickrun.v2.sh which now uses spoa which seems to be a better option for consensus calling.
best,
kevin

**Comment by alicecsq** [comment]
Thank you for your immediate response. 
I did add JAVA_HOME and re-run it. In the screenshot, I left it out intentionally just to see which poa is being executed. It is using this `/usr/bin/poa`, which may not be the correct one because it stalled at ComputeConsensus while running `./quickrun.sh`

When I tried `java -jar -Xmx40g ~/software/sicelore/Jar/Sicelore-1.0.jar ComputeConsensus T=36 I=GEUS10xAttributes.umifound.bam_2.bam O=molecules.chr2.fa TMPDIR=tmp/`
it stalled. It can parse the bam file. 
INFO    2021-04-21 13:38:13     LongreadParser          start...
INFO    2021-04-21 13:38:27     LongreadParser          end...
INFO    2021-04-21 13:38:27     LongreadParser          Total SAMrecords        [479840]
INFO    2021-04-21 13:38:27     LongreadParser          SAMrecords valid        [170711]
INFO    2021-04-21 13:38:27     LongreadParser          SAMrecords unvalid      [309129]
INFO    2021-04-21 13:38:27     LongreadParser          SAMrecords mapqv=0      [11647]
INFO    2021-04-21 13:38:27     LongreadParser          SAMrecords no gene      [0]
INFO    2021-04-21 13:38:27     LongreadParser          SAMrecords no UMI       [183569]
INFO    2021-04-21 13:38:27     LongreadParser          SAMrecords chimeria     [41049]
INFO    2021-04-21 13:38:27     LongreadParser          SAMrecords reversed     [0]
INFO    2021-04-21 13:38:27     LongreadParser          Total reads             [170711]
INFO    2021-04-21 13:38:27     LongreadParser          Total reads multiSAM    [0]
INFO    2021-04-21 13:38:27     MoleculeDataset         MoleculeDataset init start...
INFO    2021-04-21 13:38:28     MoleculeDataset         Total molecules         128714
INFO    2021-04-21 13:38:28     MoleculeDataset         Total molecule reads            170711
INFO    2021-04-21 13:38:28     MoleculeDataset         Total molecule multiIG  55
INFO    2021-04-21 13:38:28     MoleculeDataset         Calling consensus start with 36 threads
INFO    2021-04-21 13:38:28     MoleculeDataset         Total consensus to compute      128714


I have been using 'Jar/Sicelore-1.0.jar' until ComputeConsensus step. 

I tried `./quickrun.v2.sh`and it worked. 
So, I tried 'Jar/Sicelore-2.0.jar' with my own data using 
`java -jar -Xmx40g ~/software/sicelore/Jar/Sicelore-2.0.jar ComputeConsensus T=36 I=GEUS10xAttributes.umifound.bam_2.bam O=molecules.chr2.fq TMPDIR=tmp/ `

And, I got this following error. 

INFO    2021-04-21 13:31:42     LongreadParser          start...
java.lang.NullPointerException
        at org.ipmc.sicelore.utils.LongreadRecord.fromSAMRecord(LongreadRecord.java:91)
        at org.ipmc.sicelore.utils.LongreadParser.parseSAMRecord(LongreadParser.java:100)
        at org.ipmc.sicelore.utils.LongreadParser.<init>(LongreadParser.java:62)
        at org.ipmc.sicelore.programs.ComputeConsensus.doWork(ComputeConsensus.java:100)
        at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:277)
        at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:103)
        at org.ipmc.sicelore.cmdline.SiCeLoReMain.main(SiCeLoReMain.java:40)
INFO    2021-04-21 13:31:42     LongreadParser          end...
INFO    2021-04-21 13:31:42     LongreadParser          Total SAMrecords        3
INFO    2021-04-21 13:31:42     LongreadParser          SAMrecords valid        2
INFO    2021-04-21 13:31:42     LongreadParser          SAMrecords unvalid      1
INFO    2021-04-21 13:31:42     LongreadParser          SAMrecords mapqv=0      0
INFO    2021-04-21 13:31:42     LongreadParser          SAMrecords no gene      0
INFO    2021-04-21 13:31:42     LongreadParser          SAMrecords no UMI       1
INFO    2021-04-21 13:31:42     LongreadParser          SAMrecords chimeria     0
INFO    2021-04-21 13:31:42     LongreadParser          SAMrecords reversed     0
INFO    2021-04-21 13:31:42     LongreadParser          Total reads             2
INFO    2021-04-21 13:31:42     LongreadParser          Total reads multiSAM    0
INFO    2021-04-21 13:31:42     MoleculeDataset         MoleculeDataset init start...
INFO    2021-04-21 13:31:42     MoleculeDataset         Total molecules         1
INFO    2021-04-21 13:31:42     MoleculeDataset         Total molecule reads    2
INFO    2021-04-21 13:31:42     MoleculeDataset         Total molecule multiIG  0
INFO    2021-04-21 13:31:42     MoleculeDataset         Calling consensus start with 36 threads
INFO    2021-04-21 13:31:42     MoleculeDataset         Total consensus to compute      1
[Wed Apr 21 13:31:42 EDT 2021] org.ipmc.sicelore.programs.ComputeConsensus done. Elapsed time: 0.00 minutes.
Runtime.totalMemory()=2058354688

What is causing this NullPointerException?
Thank you!

Alice
